espnet/asr/pytorch_backend/asr.py

```diff
@@ -302,28 +275,46 @@ def train(args):
         mtl_mode = 'mtl'
         logging.info('Multitask learning mode')
 
-    asr_model, mt_model = None, None
-    # Initialize encoder with pre-trained ASR encoder
-    if args.asr_model:
-        asr_model, _ = load_trained_model(args.asr_model)
-        assert isinstance(asr_model, ASRInterface)
-
-    # Initialize decoder with pre-trained MT decoder
-    if args.mt_model:
-        mt_model, _ = load_trained_model(args.mt_model)
-        assert isinstance(mt_model, MTInterface)
+    if args.frontend_module is not None:
+        # The concrete class for Fronend is here: espnet.nets.pytorch_backend.frontend_asr
+        frontend_class = dynamic_import(args.frontend_module)
+        assert issubclass(frontend_class, FrontendASRInterface), frontend_class
+        # Wrap model by frontend_class.
+        frontend_asr = frontend_class(idim, args)
+        _idim = frontend_asr.featdim
+    else:
+        frontend_asr = None
+        _idim = idim
```

Comment:
Maybe this part looks complicated, but I think it's better to refactor it with the other part as a whole.
`asr.py` has too many bad coding, for example, the accessing  to the not interface attribute.

```
reporter = model.reporter
```

However, many of them come from chainer trainer restriction, so @ShigekiKarita 's Trainer is desired to be merged quickly.